packages/gcf-utils/test/logging-octokit-plugin.ts

```diff
@@ -0,0 +1,373 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {describe, beforeEach, it} from 'mocha';
+import {Octokit} from '@octokit/rest';
+import assert from 'assert';
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const LoggingOctokitPlugin = require('../src/logging-octokit-plugin.js');
+
+interface LogStatement {
+  [key: string]: string | number | LogStatement;
+}
+
+class MockLogger {
+  lastLogData: LogStatement | undefined;
+  metric(json: LogStatement) {
+    this.lastLogData = json;
+  }
+}
+
+describe('Logging-Octokit-Plugin', () => {
+  let loggingOctokit: Octokit;
+  let logger: MockLogger;
+
+  beforeEach(() => {
+    const LoggingOctokit = Octokit.plugin(LoggingOctokitPlugin);
+    logger = new MockLogger();
+    loggingOctokit = new LoggingOctokit({customLogger: logger});
+  });
+
+  it('logs information for issues.addLabels', () => {
+    loggingOctokit.issues
+      .addLabels({
+        owner: 'fooOwner',
+        issue_number: 2,
+        repo: 'barRepo',
+        labels: ['a', 'b'],
+      })
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      .catch(e => {
+        /**ignore rejected promise**/
+      })
+      .finally(() => {
+        const expected = {
+          action: {
+            type: 'ISSUE_ADD_LABELS',
+            value: 'a,b',
+            destination_repo: {
+              repo_name: 'barRepo',
+              owner: 'fooOwner',
+            },
+            destination_object: {
+              object_type: 'ISSUE',
+              object_id: 2,
+            },
+          },
+        };
+        const actual = logger.lastLogData;
+        assert.deepEqual(actual, expected);
+      });
+  });
+
+  it('logs information for issues.createComment', () => {
+    loggingOctokit.issues
+      .createComment({
+        owner: 'fooOwner',
+        issue_number: 2,
+        repo: 'barRepo',
+        body: 'comment body',
+      })
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      .catch(e => {
+        /**ignore rejected promise**/
+      })
+      .finally(() => {
+        const expected = {
+          action: {
+            type: 'ISSUE_CREATE_COMMENT',
+            value: 'comment body',
+            destination_repo: {
+              repo_name: 'barRepo',
+              owner: 'fooOwner',
+            },
+            destination_object: {
+              object_type: 'ISSUE',
+              object_id: 2,
+            },
+          },
+        };
+        const actual = logger.lastLogData;
+        assert.deepEqual(actual, expected);
+      });
+  });
+
+  it('logs information for issues.createLabel', () => {
+    loggingOctokit.issues
+      .createLabel({
+        owner: 'fooOwner',
+        repo: 'barRepo',
+        name: 'labelName',
+        color: 'blue',
+      })
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      .catch(e => {
```

Comment:
feel free to omit the catch if we're not using it (_I'm guessing this was used during debugging?_)

Reply:
The catch here is for catching and ignoring the failed call to Octokit (it fails because the Octokit is not authenticated). If I omit it then it throws errors in the test (uncaught rejected promise) and it will fail the test.

Reply:
Should we rethrow if it's not a rejected promise? If something else changes that throws and exception, this test may not catch it.